app/controllers/data_explorer/query_controller.rb

```diff
@@ -0,0 +1,236 @@
+class DataExplorer::QueryController < ::ApplicationController
+  requires_plugin DataExplorer.plugin_name
+
+  before_action :check_enabled
+  before_action :set_group, only: %i(group_reports_index group_reports_show group_reports_run)
+  before_action :set_query, only: %i(group_reports_show group_reports_run)
+
+  attr_reader :group, :query
+
+  def check_enabled
+    raise Discourse::NotFound unless SiteSetting.data_explorer_enabled?
+  end
+
+  def set_group
+    @group = Group.find_by(name: params["group_name"])
+  end
+
+  def set_query
+    @query = DataExplorer::Query.find(params[:id])
```

Comment:
`find` will raise an error if the `params[:id]` is invalid. Do we want to raise an error here? Otherwise, it'll probably be better to raise a `Discourse::InvalidParameters` error or maybe `Discourse::NotFound.`